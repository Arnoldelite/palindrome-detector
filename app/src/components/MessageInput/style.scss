.message-input {
  display: flex;
  flex-direction: row;
  width: 80%;
  margin: 20px;
  display: flex;
  justify-content: center;
  align-items: center;

  &__save {
    margin-left: 10px;
    justify-content: center;
  }
}
