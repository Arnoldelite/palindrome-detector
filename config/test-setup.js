const shell = require('shelljs')

module.exports = async () => {
  try {
    await shell.exec('npm run mongodb')
  } catch (e) {
    console.log(e)
  }
}
