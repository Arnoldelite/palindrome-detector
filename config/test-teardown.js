const shell = require('shelljs')

module.exports = async () => {
  try {
    await shell.exec('docker-compose kill')
  } catch (e) {
    console.log(e)
  }
}
