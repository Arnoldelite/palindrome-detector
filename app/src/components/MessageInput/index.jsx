import React from 'react'
import PropTypes from 'prop-types'
import { observer } from 'mobx-react'
import { Input, Button, Icon } from 'antd'
import { store } from '../..'
import { save } from '../../actions'
import './style.scss'

@observer
class MessageInput extends React.Component {
  state = {
    objectId: null,
    content: '',
  }

  componentDidMount() {
    this.setMessage(this.props.message)
  }

  saveMessage() {
    if (this.state.content) {
      const { objectId, content } = this.state
      save({ objectId, content })
      store.showEditModal(false)
      this.setState({ objectId: null, content: '' })
    }
  }

  setMessage = message => this.setState(message)

  render() {
    return (
      <div className="message-input">
        <Input
          placeholder="Enter message"
          allowClear
          value={this.state.content}
          onChange={e => this.setMessage({ content: e.target.value })}
          onPressEnter={() => this.saveMessage()}
        />
        {/* <Button className="message-input__save" type="primary" onClick={() => this.saveMessage()}>
          Send <Icon type="right" />
        </Button> */}
      </div>
    )
  }
}

MessageInput.propTypes = {
  message: PropTypes.object,
}

MessageInput.defaultProps = {
  message: {},
}

export default MessageInput
