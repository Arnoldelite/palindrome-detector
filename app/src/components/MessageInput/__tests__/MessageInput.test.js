import React from 'react'
// import Parse from 'parse/node'
import { shallow } from 'enzyme'
// import Adapter from 'enzyme-adapter-react-16'
import MessageInput from '..'
import { shallowToJson } from 'enzyme-to-json'
import '../../../../../config/jest/enzyme.config'

describe('<MessageInput />', () => {
  beforeEach(() => {
    // configure({ adapter: new Adapter() })
  })
  test('renders <MessageInput /> component', () => {
    const wrapper = shallow(<MessageInput />)
    // console.log("wrapper", wrapper.instance().state)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
