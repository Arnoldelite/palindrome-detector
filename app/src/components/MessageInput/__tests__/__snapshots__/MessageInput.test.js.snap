// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders <MessageInput /> component 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="bottom"
  title="enter word or phrase to be checked"
  transitionName="zoom-big-fast"
>
  <div
    className="message-input"
  >
    <Input
      disabled={false}
      onChange={[Function]}
      onPressEnter={[Function]}
      placeholder="Enter message"
      size="large"
      type="text"
      value=""
    />
  </div>
</Tooltip>
`;
